### PR TITLE
i18n-ed dashboard and SurveyListTable

### PR DESCRIPTION
**Pull request by sgiaccio**

### CODE REVIEW

File: webapp/loggedin/appSideBar/appSideBarFooter.js [hunk 1]
```diff
@@ -2,41 +2,36 @@ import React from 'react'
...
-        </a>
-      </div>
-    )
-  }
+      <a className="btn btn-of-link app-sidebar__btn-of"
+         href="http://www.openforis.org"
+         target="_blank">
+        {
+          opened
+            ? i18n.t('sidebar.openForis') : i18n.t('sidebar.openForisShort')
```
Comment (fefrain) [inline webapp/loggedin/appSideBar/appSideBarFooter.js]: can the ternary operator "opened ? : " go inside i18n.t ? 
e.g. i18n.t(opened ? 'sidebar.openForis' : 'sidebar.openForisShort') or even i18n.t(`sidebar.openForis${opened ? 'Short' : ''}`)

File: common/i18n/resources/en.js [hunk 2]
```diff
@@ -1,11 +1,40 @@
 module.exports = {
+  common: {
+    name: 'name',
+    cancel: 'cancel',
+    delete: 'delete',
+    cantUndoWarning: 'This operation cannot be undone!',
```
Comment (minotogna) [inline common/i18n/resources/en.js]: Let's put this key where it's used. We'll move it to general, when it'll be used more than once?

File: webapp/loggedin/modules/home/dashboard/surveyInfo/deleteSurveyDialog.js [hunk 3]
```diff
@@ -9,42 +12,47 @@ import {
...
 
   confirmNameOnChange (event) {
-    this.setState({confirmName: event.target.value})
+    this.setState({ confirmName: event.target.value })
   }
 
   render () {
-    const {surveyName, onDelete, onCancel} = this.props
+    const { surveyName, onDelete, onCancel } = this.props
+    const { i18n } = this.context
```
Comment (minotogna) [inline webapp/loggedin/modules/home/dashboard/surveyInfo/deleteSurveyDialog.js]: Can we use useContext here as well?

File: webapp/loggedin/modules/home/dashboard/surveyInfo/surveyInfo.js [hunk 4]
```diff
@@ -37,6 +39,7 @@ class SurveyInfo extends React.Component {
       publishSurvey, deleteSurvey
     } = this.props
     const { showDeleteDialog } = this.state
+    const { i18n } = this.context
```
Comment (minotogna) [inline webapp/loggedin/modules/home/dashboard/surveyInfo/surveyInfo.js]: Can we use useContext here as well?

File: common/i18n/resources/en.js (hunk 2 above)
Comment (sgiaccio) [reply common/i18n/resources/en.js]: I checked, they are used more than once but haven't gone yet thought the other modules where they are used

File: webapp/loggedin/appSideBar/appSideBarFooter.js (hunk 1 above)
Comment (sgiaccio) [reply webapp/loggedin/appSideBar/appSideBarFooter.js]: yes, moving it inside i18n

File: webapp/loggedin/modules/home/dashboard/surveyInfo/deleteSurveyDialog.js (hunk 3 above)
Comment (sgiaccio) [reply webapp/loggedin/modules/home/dashboard/surveyInfo/deleteSurveyDialog.js]: Apparently hooks don't work inside classes. We can try switch to a functional component and use the useState hook

File: webapp/loggedin/modules/home/dashboard/surveyInfo/surveyInfo.js (hunk 4 above)
Comment (sgiaccio) [reply webapp/loggedin/modules/home/dashboard/surveyInfo/surveyInfo.js]: Same as above

File: webapp/loggedin/modules/home/dashboard/surveyInfo/deleteSurveyDialog.js (hunk 3 above)
Comment (minotogna) [reply webapp/loggedin/modules/home/dashboard/surveyInfo/deleteSurveyDialog.js]: yes.

File: common/survey/_internal/surveyInfo.js [hunk 5]
```diff
@@ -84,6 +84,14 @@ const hasCollectReportIssues = R.pipe(
 
 const isFromCollect = R.pipe(getCollectUri, R.isNil, R.not)
 
+const getLanguage = preferred => surveyInfo => {
```
Comment (minotogna) [inline common/survey/_internal/surveyInfo.js]: can ```preferred``` be called something else? It's not clear what it refers to. maybe ```preferredLang``` ?

File: webapp/loggedin/modules/home/dashboard/surveyInfo/deleteSurveyDialog.js [hunk 6]
```diff
@@ -9,70 +12,66 @@ import {
...
 
-export default class DeleteSurveyDialog extends React.Component {
-  constructor (props) {
-    super(props)
-
-    this.state = {confirmName: ''}
-    this.confirmNameOnChange = this.confirmNameOnChange.bind(this)
-    this.confirmNameRef = React.createRef()
-  }
+const confirmNameRef = React.createRef()
```
Comment (minotogna) [inline webapp/loggedin/modules/home/dashboard/surveyInfo/deleteSurveyDialog.js]: it's preferred to have the variable in the scope of where it's used. in this case inside the component. Or do you use it outside it?

File: common/survey/_internal/surveyInfo.js (hunk 5 above)
Comment (sgiaccio) [reply common/survey/_internal/surveyInfo.js]: Renamed

File: webapp/loggedin/modules/home/dashboard/surveyInfo/deleteSurveyDialog.js (hunk 6 above)
Comment (sgiaccio) [reply webapp/loggedin/modules/home/dashboard/surveyInfo/deleteSurveyDialog.js]: Moved inside function

**Pull request merged**